pennylane/beta/qnode.py

```diff
@@ -0,0 +1,569 @@
+            the maximum number of derivatives to support. Increasing this value allows
+            for higher order derivatives to be extracted, at the cost of additional
+            (classical) computational overhead during the backwards pass.
+
+    Keyword Args:
+        **kwargs: Any additional keyword arguments provided are passed to the differentiation
+            method. Please refer to the :mod:`qml.gradients <.gradients>` module for details
+            on supported options for your chosen gradient transform.
+
+    **Example**
+
+    QNodes can be created by decorating a quantum function:
+
+    >>> dev = qml.device("default.qubit", wires=1)
+    >>> @qml.beta.qnode(dev)
+    ... def circuit(x):
+    ...     qml.RX(x, wires=0)
+    ...     return expval(qml.PauliZ(0))
+
+    or by instantiating the class directly:
+
+    >>> def circuit(x):
+    ...     qml.RX(x, wires=0)
+    ...     return expval(qml.PauliZ(0))
+    >>> dev = qml.device("default.qubit", wires=1)
+    >>> qnode = qml.beta.QNode(circuit, dev)
+    """
+
+    def __init__(
+        self,
+        func,
+        device,
+        interface="autograd",
+        diff_method="best",
+        max_expansion=10,
+        mode="best",
+        cache=True,
+        cachesize=10000,
+        max_diff=1,
+        **gradient_kwargs,
+    ):
+        if interface not in SUPPORTED_INTERFACES:
+            raise qml.QuantumFunctionError(
+                f"Unknown interface {interface}. Interface must be "
+                f"one of {SUPPORTED_INTERFACES}."
+            )
+
+        if not isinstance(device, Device):
+            raise qml.QuantumFunctionError(
+                "Invalid device. Device must be a valid PennyLane device."
+            )
+
+        if "shots" in inspect.signature(func).parameters:
+            warnings.warn(
+                "Detected 'shots' as an argument to the given quantum function. "
+                "The 'shots' argument name is reserved for overriding the number of shots "
+                "taken by the device. Its use outside of this context should be avoided.",
+                UserWarning,
+            )
+            self._qfunc_uses_shots_arg = True
+        else:
+            self._qfunc_uses_shots_arg = False
+
+        # input arguments
+        self.func = func
+        self.device = device
+        self._interface = interface
+        self.diff_method = diff_method
+        self.max_expansion = max_expansion
+
+        # execution keyword arguments
+        self.execute_kwargs = {
+            "mode": mode,
+            "cache": cache,
+            "cachesize": cachesize,
+            "max_diff": max_diff,
+        }
+
+        # internal data attributes
+        self._tape = None
+        self._qfunc_output = None
+        self._gradient_kwargs = gradient_kwargs
+        self._original_device = device
+        self.gradient_fn = None
+        self.gradient_kwargs = None

```

Comment:
Wait a sec, `_gradient_kwargs` and `gradient_kwargs` will be different after this, how come?

```
self._gradient_kwargs = gradient_kwargs
self.gradient_kwargs = None
```

Reply:
Good catch. `_gradient_kwargs` are the _users_ gradient keyword args. `gradient_kwargs` are the gradient keyword args _actually_ sent by the QNode after validation. We need to retain the original users kwargs, just in case validation needs to be re-done.

I think it might make sense to rename it: `_gradient_kwargs -> _user_gradient_kwargs`?

Reply:
Yes I think that'd be great :+1: